### PR TITLE
support running test_mobile_profiler with buck1/buck2 and OSS

### PR DESCRIPTION
**Pull request by dagitses**
Summary:
Internally we are switching to a new version of buck, but we also must
keep this working in OSS.

Test Plan: Rely on CI.

Differential Revision: D41270673

### CODE REVIEW

**Review by r-barnes — APPROVED** [review]

**Review by osalpekar — APPROVED** [review]

**Review by malfet — APPROVED** [review]

File: test/cpp/lite_interpreter_runtime/resources.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace torch {
+namespace testing {
+
+/// Gets the path to the resource identified by name.
+///
+/// @param name identifies a resource, relative path starting from the
+///             repo root
+inline auto getResourcePath(std::string name)
+    -> std::experimental::filesystem::path {
```
Comment (malfet) [inline test/cpp/lite_interpreter_runtime/resources.h:14]: Isn't it C++17 feature (and we still stuck with C++14 compatibility?)

Comment (dagitses) [reply test/cpp/lite_interpreter_runtime/resources.h:14]: Without the experimental it would be a C++17 feature, yes.

**Comment by dagitses** [comment]
> @pytorchbot revert -m "Broke trunk / macos-12-py3-x86-64-lite-interpreter / build" -c weird

d'oh! Does this job not run on PRs?

**Pull request closed without merging**